### ISSUE TITLE
Reset ic value when disconnection happens

### ISSUE THREAD

**Issue by wengxt** [H-M-H/xcb-imdkit-rs#5]
I didn't see any usage of this callback https://github.com/fcitx/xcb-imdkit/blob/30e2f16f9a8b0e338e25ce5e3643809a07ad41f0/src/imclient.h#L80
so I'd assume there's bug here.

This callback fires when input method itself quits. In that case, the ic value will not be a valid value anymore. and the process_event should simply let all the key event flow to application directly.

Coming from this bug:
https://github.com/wez/wezterm/issues/2819

Basically I think what happens is that, if input method itself is ever connected and successfully create ic once, the self.ic will not be None event after disconnection, which makes this code https://github.com/H-M-H/xcb-imdkit-rs/blob/master/src/lib.rs#L438 continuously forwarding event to input method even when it's not live anymore.

IMHO, the fix should be "reset self.ic to empty when disconnected fires".

**Comment by wengxt** [comment]
I think it's partially my fault that didn't make the demo app at https://github.com/fcitx/xcb-imdkit/blob/master/test/client_demo.c to consider this case. So I also updated the demo code to consider this, see:
https://github.com/fcitx/xcb-imdkit/commit/7e326359dc7e94bae97e29eb89229af266caf68c
